components/pagination/pagination.ts

```diff
@@ -0,0 +1,166 @@
+import { Component, Input, Output, OnChanges, SimpleChanges, EventEmitter, HostBinding } from "@angular/core";
+
+@Component({
+    selector: "sui-pagination",
+    template: `
+<a *ngIf="hasBoundaryLinks" class="item"  (click)="setPage(1)" [class.disabled]="page===1">
+    <span><i class="angle double left icon"></i></span>
+</a>
+<a *ngIf="hasNavigation" class="item" (click)="setPage(page-1)" [class.disabled]="!hasPrevious()">
+    <span><i class="angle left icon"></i></span>
+</a>
+<a *ngFor="let p of pages" class="item" [class.active]="p===page" (click)="setPage(p)">
+    {{ p }}
+</a>
+<a *ngIf="hasNavigation" class="item" (click)="setPage(page+1)" [class.disabled]="!hasNext()">
+    <span><i class="angle right icon"></i></span>
+</a>
+<a *ngIf="hasBoundaryLinks" class="item"  (click)="setPage(pageCount)" [class.disabled]="page===pageCount">
+    <span><i class="angle double right icon"></i></span>
+</a>
+`
+})
+export class SuiPagination implements OnChanges {
+
+    @HostBinding("class.ui")
+    @HostBinding("class.pagination")
+    @HostBinding("class.menu")
+    private _paginationClasses:boolean;
+
+    // Public members
+    public pageCount:number;
+
+    @Output()
+    public pageChange:EventEmitter<number>;
+
+    // Private members
+    private _maxSize?:number;
+    private _collectionSize:number;
+    private _pageSize:number;
+    private _page:number;
+    private _hasBoundaryLinks:boolean;
+    private _hasNavigation:boolean;
+
+    @Input()
+    public get maxSize():number {
+        return this._maxSize;
+    }
+
+    public set maxSize(value:number) {
+        this._maxSize = (value !== undefined) ? Math.max(value, 1) : undefined;
+    }
+
+    @Input()
+    public get collectionSize():number {
+        return this._collectionSize;
+    }
+
+    public set collectionSize(value:number) {
+        this._collectionSize = Math.max(value, 0);
+    }
+
+    @Input()
+    public get pageSize():number {
+        return this._pageSize;
+    }
+
+    public set pageSize(value:number) {
+        this._pageSize = value;
+    }
+
+    @Input()
+    public get hasNavigation():boolean {
+        return this._hasNavigation || this._maxSize < this.pageCount;
```

Comment:
Since comparison with `null` produces different results to `undefined`, perhaps split this out into an if statement with a null check for a bit more readability?